**Quiz 1**

'''
Design a classification problem and provide your model's synthetic prediction results. Use appropriate metrics and explain the results.
 
Create a notebook and use Sklearn with appropriate metrics to measure the performances and also discuss the results.
*Note: You need to come up with either binary or multi-class classification problem
*Note: You don't need to train a model for this quiz. You only come with your true and prediction list.
Example:
Problem: Spam Detection
True Labels:       [1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0]
Predication:       [1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0]
 
Please just provide us with your Github URL'''

My classification problem: 

  -Binary problem
  -is an individual a United States citizen


In [6]:
from sklearn.metrics import classification_report

y_true = [0, 1, 1, 1, 1,0,1,0,0]
y_prediction = [0, 0, 1, 1, 1,1,1,1,0]
target_names = ['citizen','not a citizen']

print(classification_report(y_true, y_prediction, target_names=target_names))

               precision    recall  f1-score   support

      citizen       0.67      0.50      0.57         4
not a citizen       0.67      0.80      0.73         5

     accuracy                           0.67         9
    macro avg       0.67      0.65      0.65         9
 weighted avg       0.67      0.67      0.66         9

0.6666666666666666


the results and performances of the system(when run) result in:

               precision    recall  f1-score   support

      citizen       0.67      0.50      0.57         4
not a citizen       0.67      0.80      0.73         5

     accuracy                           0.67         9
    macro avg       0.67      0.65      0.65         9
 weighted avg       0.67      0.67      0.66         9

 These results show that there is only a 67% accuracy to the test, meaning that the test is not too accurate for determining if a person is a citizen or not. Precision and recall combine to give an F1 score. The f1 score can also be seen as a way to test how accurate the test is. Thus, the test is better at determining if the person is not a citizen than determining if it is a citizen.

**Quiz 2**

I can't find it right now